src/vclock.rs

```diff
@@ -102,6 +103,15 @@ impl<A: Actor> Causal<A> for VClock<A> {
             }
         }
     }
+
+    fn read_ctx(&self) -> ReadCtx<(), A> {
+        let clock = self.clone();
+        ReadCtx {
+            add_clock: clock.clone(),
+            rm_clock: clock,
+            val: (),
+        }
+    }
```

Comment:
Yeah I think you're right, Causal isn't the right place for the `read_ctx`. Doesn't really make sense here to have a ReadCtx over a VClock.